c7n/resources/elb.py

```diff
@@ -106,6 +108,34 @@ def get_resources(self, resource_ids):
                 return []
             raise
 
+    def add_tags(self, client, elbs):
+        """
+        Adds the tags to the list of elbs
+        """
+        elb_names = [elb['LoadBalancerName'] for elb in elbs]
+        names_to_tags = {}
+
+        # max 20 ELBs per call
+        chunks = [elb_names[x:x+20] for x in xrange(0, len(elb_names), 20)]
+        fn = partial(self.process_tags, client=client)
+        with self.executor_factory(max_workers=3) as w:
+            tags = w.map(fn, chunks)
```

Comment:
there's a corner case here that makes it preferential to use the submit syntax instead of map, namely an elb being removed between the query of the elbs to querying its tags would fail the entire policy instead of the batch.